docs/topics/typecasts.md

```diff
@@ -209,7 +365,7 @@ fun main() {
 ```
 {kotlin-runnable="true" kotlin-min-compiler-version="2.0" id="kotlin-smart-casts-exception-handling"}
 
-### Smart cast prerequisites
+#### Smart cast prerequisites
 
 > Note that smart casts work only when the compiler can guarantee that the variable won't change between the check and its usage.
```

Comment:
I know it's not a part of your changes, just caught my eye that a note starts with "Note that" and has a warning color. Also, the table is simple enough to be a MD table

Reply:
I turned the note into just normal paragraph text. As for the table, the problem is that a normal markdown table highlights the first row. So if we want to keep that style we need to stick to the XML. WDYT?